### PR TITLE
Verification

### PR DESCRIPTION
**Pull request by bohatermateusz**
Code refactorization and optimizations

### CODE REVIEW

File: InterfacesHub/IReader.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿namespace InterfacesHub
+{
+    public interface IReader
+    {
+        bool Read();
```
Comment (LukaszRozmej) [inline InterfacesHub/IReader.cs]: What is the point of this method?

File: Interview.Tests/CSVReaderWriterTests.cs [hunk 2]
```diff
@@ -4,33 +4,40 @@
...
+            const string column2 = "column2";
+
+            while (_instance.Read())
             {
                 columns.Add((column1, column2));
             }
-            
+
             Assert.That(columns.Count, Is.EqualTo(1000));
+
```
Comment (LukaszRozmej) [inline Interview.Tests/CSVReaderWriterTests.cs]: Doesn't actually supply columns collection with meaningful values

**Review by LukaszRozmej — CHANGES_REQUESTED** [review]
Pros:
Good high level separation of concerns. Making smaller clearer classes with single responsibility.

Cons (big ones):
No review supplied in CsvReaderWriter_review.cs,
Read method doesn't do anything useful,
Low level code wasn't fixed

**Comment by bohatermateusz** [comment]
> Pros:
> Good high level separation of concerns. Making smaller clearer classes with single responsibility.
> 
> Cons (big ones):
> No review supplied in CsvReaderWriter_review.cs,
> Read method doesn't do anything useful,
> Low level code wasn't fixed

Comments to CsvReaderWriter_review, was provided here:
https://github.com/bohatermateusz/MyTechTask/commit/aab3eafa4a2243dca6f7b6f53a49b26017130661

You are totally right, Read method doesn't do anything special and it is not doing this for which it was created. I have assumed that code which was written by intern is working(including low level code) and my job is to focus on refactorization. I will fix that.

File: InterfacesHub/IReader.cs (hunk 1 above)
Comment (bohatermateusz) [reply InterfacesHub/IReader.cs]: Interface might be fine. Implementation of Interface is wrong (which doesn't do anything special)

File: Interview.Tests/CSVReaderWriterTests.cs (hunk 2 above)
Comment (bohatermateusz) [reply Interview.Tests/CSVReaderWriterTests.cs]: True

**Comment by LukaszRozmej** [comment]
bool Read(out string column1, out string column2) was working

**Comment by bohatermateusz** [comment]
> bool Read(out string column1, out string column2) was working

Ok, I have change both of them to tuple values

**Pull request merged**